### PR TITLE
upgrade go-related bins to latest

### PR DESCRIPTION
**Pull request by Warashi**
## TL;DR
Upgrade go-related bins to latest

## Details
protoc-gen-go's install path changed to `google.golang.org/protobuf/cmd/protoc-gen-go`
ref: https://developers.google.com/protocol-buffers/docs/gotutorial

gogo/protobuf has some more faster binaries
ref: https://github.com/gogo/protobuf#more-speed-and-more-generated-code

micro/protobuf install path changed to `github.com/micro/micro/v3/cmd/protoc-gen-micro@master `
ref 1: https://github.com/micro/protobuf
ref 2: https://github.com/micro/protoc-gen-micro
ref 3: https://github.com/micro/micro/tree/master/cmd/protoc-gen-micro

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -61,15 +61,18 @@ RUN go get -u google.golang.org/grpc
...
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogo
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofast
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofaster
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogoslick
 
 RUN go get -u github.com/ckaznocha/protoc-gen-lint
 RUN go get -u github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
 
-RUN go get -u github.com/micro/protobuf/protoc-gen-go
+RUN go get -u github.com/micro/micro/v3/cmd/protoc-gen-micro@master
```
Comment (vtopc) [inline Dockerfile:75]: this is not working:
```
go: cannot use path@version syntax in GOPATH mode
```

Comment (Warashi) [reply Dockerfile]: sorry, fixed by https://github.com/namely/docker-protoc/pull/206/commits/91f860d7d012712df268e692729f54df5e71e6ea

File: Dockerfile [hunk 2]
```diff
@@ -61,15 +61,18 @@ RUN go get -u google.golang.org/grpc
...
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogo
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofast
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofaster
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogoslick
 
 RUN go get -u github.com/ckaznocha/protoc-gen-lint
 RUN go get -u github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
 
-RUN go get -u github.com/micro/protobuf/protoc-gen-go
+RUN go get -u github.com/micro/micro/v3/cmd/protoc-gen-micro
```
Comment (vtopc) [inline Dockerfile:75]: Could you test it locally?
```
make build

package github.com/micro/micro/v3/cmd/protoc-gen-micro: cannot find package "github.com/micro/micro/v3/cmd/protoc-gen-micro" in any of:
	/usr/local/go/src/github.com/micro/micro/v3/cmd/protoc-gen-micro (from $GOROOT)
	/go/src/github.com/micro/micro/v3/cmd/protoc-gen-micro (from $GOPATH)
The command '/bin/sh -c go get -u github.com/micro/micro/v3/cmd/protoc-gen-micro' returned a non-zero code: 1
```

Comment (Warashi) [reply Dockerfile]: very sorry, fixed and tested

Comment (vtopc) [reply Dockerfile]: Awesome, thank you

File: Dockerfile [hunk 3]
```diff
@@ -59,15 +59,18 @@ RUN go get -u google.golang.org/grpc
 
 RUN go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
 RUN go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
-RUN go get -u github.com/golang/protobuf/protoc-gen-go
+RUN go get -u google.golang.org/protobuf/cmd/protoc-gen-go
```
Comment (vtopc) [inline Dockerfile:62]: So, based on
```
(1) github.com/golang/protobuf@v1.3.4 is the most recent pre-APIv2 version of APIv1.

(2) github.com/golang/protobuf@v1.4.0 is a version of APIv1 implemented in terms of APIv2. The API is the same, but the underlying implementation is backed by the new one. This version contains functions to convert between the APIv1 and APIv2 proto.Message interfaces to ease the transition between the two.

(3) google.golang.org/protobuf@v1.20.0 is APIv2. This module depends upon github.com/golang/protobuf@v1.4.0, so any program which uses APIv2 will automatically pick a version of APIv1 which integrates with it.
```
from https://blog.golang.org/protobuf-apiv2 the plan is next:
switch to (2) first, then switch to (3), and currently, master on pre-(1) as `protoc-gen-go` plugin is overwritten by the   `github.com/micro/protobuf/protoc-gen-go`(which is 5 years old)

File: Dockerfile [hunk 4]
```diff
@@ -59,15 +59,18 @@ RUN go get -u google.golang.org/grpc
...
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogo
 RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofast
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogofaster
+RUN go get -u github.com/gogo/protobuf/protoc-gen-gogoslick
 
 RUN go get -u github.com/ckaznocha/protoc-gen-lint
 RUN go get -u github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
 
-RUN go get -u github.com/micro/protobuf/protoc-gen-go
+RUN go get -u github.com/micro/micro/cmd/protoc-gen-micro
```
Comment (vtopc) [inline Dockerfile:73]: moved this into separate PR #210

**Comment by Warashi** [comment]
resolved conflicts

**Comment by ido-namely** [comment]
@Warashi I believe this has been addressed in a different pr, so I'm closing the pr for now, but let me know otherwise.

**Comment by Warashi** [comment]
@ido-namely 
`protoc-gen-go` is stil APIv1?
https://github.com/namely/docker-protoc/blob/6f1d440e14c2dad0501de66d304acf4714d70819/Dockerfile#L95-L96

**Comment by ido-namely** [comment]
That's correct, but I believe under the hood it's still the new api.
Perhaps we should strive to upgrade, but I don't think it's urgent and that using this version blocks anything (please let me know if I'm wrong).
see more info here  - https://github.com/protocolbuffers/protobuf-go/releases/tag/v1.20.0#v1.20-grpc-support

**Comment by Warashi** [comment]
I also think using this version blocks nothing, but I think that using new api is better.

Because
1. protoc-gen-go APIv1 generates go code with `github.com/golang/protobuf` as dependency
2. protoc-gen-go-grpc is already provided, and gRPC Quick start shows to use protoc-gen-go APIv2
  ref: https://grpc.io/docs/languages/go/quickstart/
